New connection settings `Connection Timeout` field

sebjulliand opened the pull request:
### Changes
Resolves #2442 and #2440.

This PR adds a new `readyTimeout` field to the connection settings (i.e. to the `ConnectionData` type). This field is described [here](https://github.com/mscdex/ssh2#client-methods).

A new `number` input has been added to the Connection/Login Settings to manage this field:
![image](https://github.com/user-attachments/assets/e2340c26-a798-46d3-80bb-0e4fb6d17879)

The `CustomUI` class has been enhanced to support the `number` input type as well as its `min` and `max` attributes.
The SSH port field has been turned into a `number` input.

### How to test this PR
1. Create a new connection with wrong credentials
2. Change the `Connection Timeout` field to 1000 (i.e. 1 second)
3. Try to connect: the connection attempt must fail after 1 second
4. Change the `Connection Timeout` field to 5000
5. Try to connect: the connection attempt must fail after 5 seconds

### Checklist
* [x] have tested my change

julesyan commented:
There is some odd behaviour that I have been able to inconsistently recreate. I set the value to 2000 (or 20,000), double checked the value. Connected to the system successfully. Disconnected, then checked the value again and it seems to have changed the timeout value to 1999 (or was 19,998). I will see if i can consistently recreate it, but so far its sporadic behaviour. 
<img width="586" alt="image" src="https://github.com/user-attachments/assets/7cda0bd1-230a-486e-a3df-bcc5699303cb" />

julesyan commented:
I can set the readyTimeout field to 0, even though it has min set to 1. It shows as red, but the save is still allowed. I checked the settings file and it does save as 0. It does not treat the timeout as 0ms, it seems to just ignore the timeout when it is 0. 
<img width="522" alt="image" src="https://github.com/user-attachments/assets/32d606e9-43f3-4b24-a99a-07c1bd6edc7c" />

In regards to the other issue I mentioned with the timeout value changing, I have been unable to recreate it more than 2 more times after 200+ tries. So it seems to be a possible one off of the string to number conversion. I wouldn't be concerned about it.

sebjulliand commented:
Thanks @julesyan !
I fixed the numeric value not being entirely checked - setting a value lesser than 1 will not allow you to save the settings.

As for the odd behavior, I think I found what the "issue" is. Spoiler alert: it's not a bug, it's a feature. 😅
It turns out that if a numeric field has the focus and your mouse pointer it on it, the value will increment/decrement if you use your mouse scrollwheel!

To try to prevent this from happening, I moved the field to the bottom of the form. Since it used to be in the middle, it was natural to use the wheel to scroll down to the `Save` button, with the risk of changing the value while doing so.

Give it a try and let me know how it goes!

sebjulliand commented:
I changed the type of the SSH port number to a "number" input, so there's a better check on the value.

julesyan reviewed (changes requested):
Everything looks good aside from if i have an invalid value for the timeout, then use the scroll or up/down arrows on the field, it does not revalidate that data, it stays as an error

sebjulliand commented:
@julesyan it should be OK now. I used a more global event to catch changes in any `input` field. Let me know how it goes!

julesyan reviewed (approved):
Looks good!

sebjulliand commented:
Much obliged @julesyan 🙏🏻
